File: api/src/main/java/api/com/br/api/domain/Midia.java
package api.com.br.api.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "midias")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class Midia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @Column(name = "nome_arquivo")
    private String nomeArquivo;

    @Column(name = "tipo_midia")
    private String tipoMidia;

    private byte[] conteudo;

    @Column(name = "criado_em")
    private LocalDateTime criadoEm;

    @Column(name = "modificado_em")
    private LocalDateTime modificadoEm;

}

File: api/src/domain/Midia.ts
import { Usuario } from './Usuario';

export class Midia {
  id: number;
  usuario_id: number;
  nome_arquivo: string;
  conteudo: Buffer;
  criado_em: Date;
  modificado_em: Date;
  usuario: Usuario;

  constructor(
    id: number,
    usuario_id: number,
    nome_arquivo: string,
    conteudo: Buffer,
    criado_em: Date,
    modificado_em: Date,
    usuario: Usuario
  ) {
    this.id = id;
    this.usuario_id = usuario_id;
    this.nome_arquivo = nome_arquivo;
    this.conteudo = conteudo;
    this.criado_em = criado_em;
    this.modificado_em = modificado_em;
    this.usuario = usuario;
  };
};
